motion/thruster_allocator/include/thruster_allocator/allocator_utils.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef VORTEX_ALLOCATOR_ALLOCATOR_UTILS_HPP
+#define VORTEX_ALLOCATOR_ALLOCATOR_UTILS_HPP
+
+#include "rclcpp/rclcpp.hpp"
+#include "vortex_msgs/msg/thruster_forces.hpp"
+#include <eigen3/Eigen/Eigen>
+#include <string>
+#include <vector>
+
+// Return true if M has any NaN or INF elements.
+template <typename Derived>
+inline bool isInvalidMatrix(const Eigen::MatrixBase<Derived> &M) {
+  bool has_nan = !(M.array() == M.array()).all();
+  bool has_inf = M.array().isInf().any();
+  return has_nan || has_inf;
+}
+
+inline void printMatrix(std::string name, const Eigen::MatrixXd &M) {
+  std::stringstream ss;
+  ss << std::endl << name << " = " << std::endl << M;
+  RCLCPP_INFO_STREAM(rclcpp::get_logger("rclcpp"), ss.str());
+}
+
+inline void printVector(std::string name, const Eigen::VectorXd &M) {
+  std::stringstream ss;
+  ss << std::endl << name << " = " << std::endl << M;
+  RCLCPP_INFO_STREAM(rclcpp::get_logger("rclcpp"), ss.str());
+}
+
+// Calculate the pseudoinverse matrix of the matrix M.
+// Return false if the calculations fails.
+inline bool calculatePseudoinverse(const Eigen::MatrixXd &M,
+                                   Eigen::MatrixXd *M_pinv) {
+  Eigen::MatrixXd pseudoinverse = M.transpose() * (M * M.transpose()).inverse();
```

Comment:
Because we don't know if it gives out the Left or Right psuedoinverse. We could rename the function to be calculateRightPseudoinverse, as that is the type of pinv we are calculating here.